src/models/ratings/ratings.service.ts

```diff
@@ -0,0 +1,443 @@
+						playersArray[i].boost,
+						playersArray[i].ratedgames,
+						playersArray[i].maxrating,
+						playersArray[i].ratingage,
+						JSON.stringify(playersArray[i].fatigue),
+						this.adjustedRating(playersArray[i], Date.now(), PARTICIPATION_CUTOFF, RATING_RETENTION, MAX_DROP, PARTICIPATION_LIMIT),
+						playersArray[i].id,
+					);
+				}
+				// always update fatiguerating
+				const updateFatigueRatingQuery = `UPDATE players SET fatiguerating=? where id=?;`;
+				await playerRunner.manager.query(updateFatigueRatingQuery, [this.adjustedRating(playersArray[i], Date.now(), PARTICIPATION_CUTOFF, RATING_RETENTION, MAX_DROP, PARTICIPATION_LIMIT), playersArray[i].id]);
+			}
+			await playerRunner.commitTransaction();
+			await gameRunner.commitTransaction();
+		} catch (error) {
+			console.error(error);
+			this.logger.error("Error processing player and game ratings, rolling back transaction. ", error);
+			playerRunner.rollbackTransaction();
+			gameRunner.rollbackTransaction();
+		} finally {
+			playerRunner.release();
+			gameRunner.release();
+			if (!has_error) {
+				this.logger.debug('Finished updating ratings');
+				await writeFile('previous.txt', lastUsedGame + ' ' + 'hash');
+			}
+		}
+	}
+
+	// returns the adjusted rating of a player
+	public adjustPlayer(
+		player: Player,
+		amount: number,
+		fairness: number,
+		fatigueFactor: number,
+		date: number,
+		BONUS_FACTOR: number,
+		BONUS_RATING: number,
+		RATING_RETENTION: number,
+		INITIAL_RATING: number,
+	): Player{
+		const bonus = Math.min(
+			Math.max(0, (fatigueFactor * amount * Math.max(player.boost, 1) * BONUS_FACTOR) / BONUS_RATING),
+			player.boost,
+		);
+		player.boost -= bonus;
+		const k =
+			10 +
+			15 * Math.pow(0.5, player.ratedgames / 200) +
+			15 * Math.pow(0.5, (player.maxrating - INITIAL_RATING) / 300);
+		player.rating += fatigueFactor * amount * k + bonus;
+		if (player.ratingage == 0) {
+			player.ratingage = date - RATING_RETENTION;
+		}
+		let participation = 20 * Math.pow(0.5, (date - player.ratingage) / RATING_RETENTION);
+		participation += fairness * fatigueFactor;
+		participation = Math.min(20, participation);
+		player.ratingage = Math.log2(participation / 20) * RATING_RETENTION + date;
+		player.ratedgames++;
+		player.maxrating = Math.max(player.maxrating, player.rating);
+		player.changed = true;
+		return player;
+	}
+
+	// returns the new fatigue value
+	public updateFatigue(player: Player, opponentID: string, gameFactor: number) {
+		const MULTIPLIER = 1 - gameFactor * 0.4;
+		// check if player fatigue is a string adn throw error
+		if (typeof player.fatigue == 'string') {
+			throw new Error('Fatigue is a string, needs to be an object');
+		}
+		for (const a in player.fatigue) {
+			player.fatigue[a] *= MULTIPLIER;
+			if (a != opponentID && player.fatigue[a] < 0.01) {
+				delete player.fatigue[a];
+			}
+		}
+		return (player.fatigue[opponentID] = (player.fatigue[opponentID] || 0) + gameFactor);
+	}
+
```

Comment:
contrasting too adjustPlayer, adjustedRating just returns the value

Reply:
I updated the comments to explain it better what it's doing.